pkg/protoconv/resources/resources.go

```diff
@@ -122,6 +123,9 @@ func newWrap(meta metav1.Object, kind, clusterID, registryOverride string) *Depl
 // SpecToPodTemplateSpec turns a top level spec into a podTemplateSpec
 func SpecToPodTemplateSpec(spec reflect.Value) (v1.PodTemplateSpec, error) {
 	templateInterface := spec.FieldByName("Template")
+	if !doesFieldExist(spec) {
```

Comment:
Do you mean `!doesFieldExist(templateInterface)`? Otherwise, this should be above the `templateInterface := spec.FieldByName("Template")` line

Reply:
Ah, good catch.